### PR TITLE
MGDOBR-547 Add  e2e test for Processor filter update

### PR DESCRIPTION
**Pull request by Srihari1192**
[MGDOBR-547](https://issues.redhat.com/browse/MGDOBR-547)

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.

</details>

### CODE REVIEW

File: integration-tests/src/test/resources/features/processor.feature [hunk 1]
```diff
@@ -30,3 +31,92 @@ Feature: Processor tests
     And the Manager metric 'manager_processor_status_change_total{status="PROVISIONING",}' count is at least 1
     And the Manager metric 'manager_processor_status_change_total{status="READY",}' count is at least 1
     And the Manager metric 'manager_processor_status_change_total{status="DELETED",}' count is at least 1
+
+  Scenario: Processor filter is updated
```
Comment (sutaakar) [inline integration-tests/src/test/resources/features/processor.feature:35]: Please move the scenario to `processor-filter.feature` to keep such tests separate from "plain" processor tests.

File: integration-tests/src/test/resources/features/processor.feature [hunk 2]
```diff
@@ -30,3 +31,92 @@ Feature: Processor tests
...
+      "name": "myProcessor",
+      "action": {
+        "parameters": {
+            "topic":  "myKafkaTopic"
+        },
+        "type": "KafkaTopic"
+      }
+    }
+    """
+    And the Processor "myProcessor" of the Bridge "mybridge" is existing with status "ready" within 3 minutes
```
Comment (sutaakar) [inline integration-tests/src/test/resources/features/processor.feature:122]: Can you please send a cloud event to check that the updated processor is actually deployed and updated filtering is applied?

Comment (Srihari1192) [reply integration-tests/src/test/resources/features/processor.feature:122]: Sure will add a step to send cloud events after filter update

File: integration-tests/src/test/resources/features/processor-filter.feature [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    "specversion": "1.0",
+    "type": "StringEquals",
+    "source": "StorageService",
+    "id": "filter-event-test",
+    "data": {
+        "name": "Hello world"
+      }
+    }
+    """
+    Then Webhook site contains request with text "${cloud-event.filter-event-test.id}" within 1 minute
```
Comment (sutaakar) [inline integration-tests/src/test/resources/features/processor-filter.feature:56]: Can you please add here also a check making sure that event not matching the filter is not published?

File: integration-tests/src/test/resources/features/processor-filter.feature [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    "specversion": "1.0",
+    "type": "StringContains",
+    "source": "StorageService",
+    "id": "filter-event-test-second",
+    "data": {
+        "name": "John"
+      }
+    }
+    """
+    Then Webhook site contains request with text "${cloud-event.filter-event-test-second.id}" within 1 minute
```
Comment (sutaakar) [inline integration-tests/src/test/resources/features/processor-filter.feature:99]: Same here.

File: integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/steps/WebhookSiteSteps.java [hunk 5]
```diff
@@ -32,4 +32,17 @@ public void webhookSiteContainsRequest(String requestText, int timeoutMinutes) {
...
+    public void webhookSiteDoesNotContainsRequest(String requestText, int timeoutMinutes) {
+        String requestTextWithoutPlaceholders = ContextResolver.resolveWithScenarioContext(context, requestText);
+        Awaitility.await()
+                .atMost(Duration.ofMinutes(timeoutMinutes))
+                .pollInterval(Duration.ofSeconds(1))
+                .untilAsserted(() -> assertThat(WebhookSiteResource.requests())
+                        .map(request -> request.getContent())
+                        .as("Searching for request containing text: '%s'",
+                                requestTextWithoutPlaceholders)
+                        .noneMatch(requestContent -> requestContent.contains(requestTextWithoutPlaceholders)));
```
[the remark this answers is not in the document]
Comment (sutaakar) [reply integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/steps/WebhookSiteSteps.java:46]: So you need a loop without timeout error as the timeout is desired state.

File: integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/steps/WebhookSiteSteps.java [hunk 6]
```diff
@@ -32,4 +32,17 @@ public void webhookSiteContainsRequest(String requestText, int timeoutMinutes) {
                                 requestTextWithoutPlaceholders)
                         .anyMatch(requestContent -> requestContent.contains(requestTextWithoutPlaceholders)));
     }
+
+    @Then("^Webhook site does not contains request with text \"([^\"]*)\" within (\\d+) (?:minute|minutes)$")
```
Comment (sutaakar) [inline integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/steps/WebhookSiteSteps.java:36]: The minute long timeout is quite long, IMHO it is enough to have 10 second timeout

**Comment by sutaakar** [comment]
@Srihari1192 I see that you test the update twice, what is the intention for such testing?
Wouldn't be enough to test just one update?

File: integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/steps/WebhookSiteSteps.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (Srihari1192) [reply integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/steps/WebhookSiteSteps.java:46]: Sure will implement accordingly

**Comment by Srihari1192** [comment]
> @Srihari1192 I see that you test the update twice, what is the intention for such testing? Wouldn't be enough to test just one update?

I added multiple update to verify all the supported filters..I think its enough to test with one filter update

**Comment by sutaakar** [comment]
Checking one filter update should be enough.

**Comment by Srihari1192** [comment]
Closing and recreated here https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/675

**Pull request closed without merging**